app/src/main/java/org/oppia/app/story/StoryViewModel.kt

```diff
@@ -31,28 +31,27 @@ class StoryViewModel @Inject constructor(
   private var internalProfileId: Int = -1
   private lateinit var topicId: String
 
+  private lateinit var storyResultLiveData: LiveData<AsyncResult<StorySummary>>
+  private lateinit var storyLiveData: LiveData<StorySummary>
+  lateinit var storyNameLiveData: LiveData<String>
+  lateinit var storyChapterLiveData: LiveData<List<StoryItemViewModel>>
+
   /** [storyId] needs to be set before any of the live data members can be accessed. */
   private lateinit var storyId: String
-  private val explorationSelectionListener = fragment as ExplorationSelectionListener
+  private lateinit var explorationSelectionListener: ExplorationSelectionListener
+  private lateinit var newFragment: Fragment
 
-  private val storyResultLiveData: LiveData<AsyncResult<StorySummary>> by lazy {
-    topicController.getStory(
+  fun setStoryResultLiveData(fragment: Fragment) {
```

Comment:
This is a rough solution I pushed so that we can discuss same in meeting easily.